lib/rules/init-declarations.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @fileoverview A rule to control the style of variable initializations.
+ * @author Colin Ihrig
+ * @copyright 2015 Colin Ihrig. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    var MODE_ALWAYS = "always",
+        MODE_NEVER = "never";
+
+    var mode = context.options[0] || MODE_ALWAYS;
+
+    //--------------------------------------------------------------------------
+    // Public API
+    //--------------------------------------------------------------------------
+
+    return {
+        "VariableDeclaration": function(node) {
+
+            var kind = node.kind,
+                declarations = node.declarations;
+
+            for (var i = 0; i < declarations.length; ++i) {
+                var initialized = declarations[i].init !== null;
+
+                if (mode === MODE_ALWAYS && !initialized) {
+                    context.report(node, "Variables should be initialized on declaration.");
```

Comment:
it would nice to have the name of the variable inside the message to help people understand what variable you are talking about.